packages/api/src/sequelize/migrations/2024-08-06_00_create-mappings-jwt.ts

```diff
@@ -0,0 +1,107 @@
+import { DataTypes } from "sequelize";
+import type { Migration } from "..";
+import * as shared from "../migrations-shared";
+
+const cxMappingTableName = "cx_mapping";
+const patientMappingTableName = "patient_mapping";
+const jwtTokenTableName = "jwt_token";
+
+// Use 'Promise.all' when changes are independent of each other
+export const up: Migration = async ({ context: queryInterface }) => {
+  await queryInterface.sequelize.transaction(async transaction => {
+    await shared.createTable(
+      queryInterface,
+      cxMappingTableName,
+      {
+        id: {
+          type: DataTypes.STRING,
+          primaryKey: true,
+          allowNull: false,
+        },
+        cxId: {
+          type: DataTypes.STRING,
+          field: "cx_id",
+          allowNull: false,
+        },
+        source: {
+          type: DataTypes.STRING,
+          field: "source",
+          allowNull: false,
+        },
+        externalId: {
+          type: DataTypes.STRING,
+          field: "external_id",
+          allowNull: false,
+        },
+      },
+      { transaction, addVersion: true }
+    );
+    await shared.createTable(
+      queryInterface,
+      patientMappingTableName,
+      {
+        id: {
+          type: DataTypes.STRING,
+          primaryKey: true,
+          allowNull: false,
+        },
+        patientId: {
+          type: DataTypes.STRING,
+          field: "patient_id",
+          allowNull: false,
+        },
+        source: {
+          type: DataTypes.STRING,
+          field: "source",
+          allowNull: false,
+        },
+        externalId: {
+          type: DataTypes.STRING,
+          field: "external_id",
+          allowNull: false,
+        },
```

Comment:
This and source, yes (if we want it to be fast). Done.

Reply:
Do we plan to list all IDs from a given source? Otherwise, we could just use `externalId` on the index, which is probably what we'll be filtering for, right? It would make is cheaper to update than a multi-column one, AFAIK.

Reply:
Two sources may have conflicting ids -- unlikely but not impossible. Both would be valid, and shouldn't interrupt each other.

Reply:
No problem, if there's more than one entry for a single externalId, the DB would easily find those and then search through them w/ the other columns. This is not a `unique` index, so no constraint is being created that requires the index to have the source, right?

Reply:
The source has to be non-null. I guess it's just a trade-off then right? multi-column index -- slower to update but faster to search, vs. the opposite.

Inserting is rare for these tables -- getting happens in the middleware, so on every request. I'd error on a slower to update but faster query path given the usage pattern.

Reply:
I *did* add a unique index on externalId / source so that you can have the same map go to two different customers or patients.

Reply:
I noticed we're adding a unique constraint with both columns, so we need them on the index. All good.